hil/api.py

```diff
@@ -420,6 +419,19 @@ def _have_attachment(nic, query):
         raise errors.BadArgumentError(
             "Channel %r, is not legal for this network." % channel)
 
+    # First, check if the switch supports not having any native networks.
+    # then check if the passed channel is native, if yes, then check in the db
+    # if there's a native network already in use.
+    switch = nic.port.owner
+    native_channel = allocator.get_native_channel()
```

Comment:
You can only use methods of the network allocator which are part of the base class `hil.network_allocator.NetworkAllocator`. The way it is not it will break for all other network allocators that don't define it.

It would actually be "ok" to put this inside the if condition below, **but** we must clearly document that there is now a dependency that **all** switches that don't have the capability to not require a native network must be used with an allocator that has the `get_native_channel()` method.

But then, I don't like that this introduces a requirement on **all** switches that don't know anything about native networks, to declare a capability that they don't require one.

Reply:
> must be used with an allocator that has the get_native_channel() method.